tasks/apply-mapping/apply-mapping.yaml

```diff
@@ -202,7 +202,10 @@ spec:
 
         if [ "$(params.failOnEmptyResult)" = "true" ] && \
           [ "$(jq '.components | length' < "${SNAPSHOT_SPEC_FILE}")" -eq 0 ]; then
-          echo "ERROR: Resulting snapshot contains 0 components"
+          echo "ERROR: Resulting snapshot contains 0 components. This means that there were 0 components present in"
+          echo "both your Snapshot and your ReleasePlanAdmission mapping. Take a look at your component names and"
+          echo "make sure that all components from the snapshot you want to release are present in the"
```

Comment:
Heh, I am actually working on a similar PR :D But isn't this incorrect? You don't necessarily need to map all components. I guess you will in most cases, but I thought it was not mandatory.

Reply:
It says all the components you want to release. I think it's ok.

Reply:
Ok, you can read it the other way around I guess. Maybe `all the components you want to release are present in the `RPA`.

Reply:
Also, can we at least print the components for easy comparison:
```sh
echo "Components in snapshot: $(-c '[.components[].name]' < "${SNAPSHOT_SPEC_FILE_ORIG}")"
echo "Components in mapping: $(-c '[.components[].name]' < "${MAPPING}")"
```

Reply:
updated

Reply:
From the unit test logs:
```
[run-task : apply-mapping] ++ jq -c '[.components[].name]' /workspace/config/snapshot_spec.json.orig
[run-task : apply-mapping] Components in snapshot: ["comp1","comp2"]
[run-task : apply-mapping] + echo 'Components in snapshot: ["comp1","comp2"]'
[run-task : apply-mapping] ++ jq -c '[.components[].name]'
[run-task : apply-mapping] Components in mapping: ["comp3","comp4"]
[run-task : apply-mapping] + echo 'Components in mapping: ["comp3","comp4"]'
[run-task : apply-mapping] + exit 1
```

Reply:
Cool, thanks.